src/book/03-guides/01-main-codebase/04-robocup-setup.mdx

```diff
@@ -5,30 +5,40 @@ title: Robocup Setup
 description: Setting up the robot for robocup
 slug: /guides/main/robocup-setup
 authors:
-    - Cameron Murtagh (@CMurtagh-LGTM)
+  - Cameron Murtagh (@CMurtagh-LGTM)
 ---
 
 ## Networking
 
-1. On the robot with a monitor and keyboard, not via `ssh`.
-2. Rename `system/nugusX/etc/systemd/network/40-wlp58s0-robocup.network`
-   to `system/nugusX/etc/systemd/network/30-wlp58s0.network`
-3. Rename `system/defualt/etc/systemd/30-wlp58s0.network` to `system/default/etc/systemd/40-wlp58s0-robocup.network`
-4. Double check that `Address`, in the file now called `30-wlp58s0.network`, is in the subnet you are connecting to,
-   this is the address of this robot.
-   Usually the subnet mask is `255.255.0.0` meaning that the first two numbers define the subnet.
-   Usually the address of the subnet is `192.168.X.X` or `10.1.X.X`.
-5. Also check that `Gateway` and the first `DNS` are correct in this file, both gateway and dns normally end with
-   `.3.1` and are in the subnet.
-6. Add the output of `wpa_passphrase SSID PASSWORD` to `system/default/etc/wpa_supplicant/wpa_supplicant-wlp58s0.conf`
-   where `SSID` is the SSID and `PASSWORD` is the password of the Wi-Fi network you are connecting to.
-   Now add `priority=3` inside the curly brackets on a new line to the entry that you just added.
-   You can use `vim` or `nano` to edit files on the robot.
-7. Run `sudo ./system_configuration`.
-8. Reboot the robot `reboot` or restart the services
-   `sudo systemctl restart wpa_supplicant@wlp58s0` and `sudo systemctl restart wpa_supplicant@wlp58s0`.
+1.  Checkout the branch for robocup and change directory to `modules/tools/SystemConfiguration/data/system`.
+2.  Rename `default/etc/systemd/30-wlp58s0.network` to `default/etc/systemd/40-wlp58s0-robocup.network`
+3.  Add the output of `wpa_passphrase SSID PASSWORD` to `default/etc/wpa_supplicant/wpa_supplicant-wlp58s0.conf`
+    where `SSID` is the SSID and `PASSWORD` is the password of the Wi-Fi network you are connecting to.
+    Now add `priority=3` inside the curly brackets on a new line to the entry that you just added.
+    The block should look like, where `HASHED_PASSWORD` is a long string of letters and numbers
+    ```
+    network={
+           ssid="SSID"
+           #psk="PASSWORD"
+           psk=HASHED_PASSWORD
+           priority=3
+    }
+    ```
```

Comment:
I still thing you should just tell people to add this block to the end of `default/etc/wpa_supplicant/wpa_supplicant-wlp58s0.conf`, they have to come open this file in an editor to add the `priority=3`line anyway, and telling them to add the whole block to the end of the file circumvents people accidentally adding the priority line to the wrong block 
```suggestion
    ```
    network={
           ssid="SSID"
           psk="PASSWORD"
           priority=3
    }
    ```
```

Reply:
Does this work? I remember trying to manually put in the info once before and it wouldn't work until I added it in with `wpa_passphrase` and had the hashed password? But maybe I did something else wrong :shrug:

Reply:
The manpages wpa_supplicant.conf specifically say
```
psk          WPA preshared key used in WPA-PSK mode.  The key is specified as
	     64	hex digits or as an 8-63 character ASCII passphrase.  ASCII
	     passphrases are dynamically converted to a	256-bit	key at runtime
	     using the network SSID, or	they can be statically converted at
	     configuration time	using the wpa_passphrase(8) utility.
```
And this forum post also says this
https://superuser.com/a/679959/1676555

so it should work just fine

Reply:
this would be better, I'll test and update it